packages/asset-swapper/src/utils/market_operation_utils/constants.ts

```diff
@@ -2041,11 +2044,20 @@ export const DEFAULT_GAS_SCHEDULE: Required<FeeSchedule> = {
         return gas;
     },
     [ERC20BridgeSource.UniswapV3]: (fillData?: FillData) => {
-        let gas = 100e3;
-        const path = (fillData as UniswapV3FillData).tokenAddressPath;
-        if (path.length > 2) {
-            gas += (path.length - 2) * 32e3; // +32k for each hop.
+        const uniFillData = fillData as UniswapV3FillData | FinalUniswapV3FillData;
+        // NOTE: This base value was heuristically chosen by looking at how much it generally
+        // underestimated gas usage
+        let gas = 34e3; // 34k base
+        if (isFinalUniswapV3FillData(uniFillData)) {
+            gas += uniFillData.gasUsed;
+        } else {
+            // Base gas usage estimate on all unique paths sorted in descending order by amount
+            const sortedPathAmounts = uniFillData.pathAmounts.sort((a, b) => b.inputAmount.comparedTo(a.inputAmount));
+            const uniqPathAmounts = _.uniqBy(sortedPathAmounts, p => p.uniswapPath);
+            const totalGasUsed = uniqPathAmounts.reduce((memo, p) => memo + p.gasUsed, 0);
+            gas += totalGasUsed;
```

Comment:
I didn't fully expect an accumulator here, can you refresh my memory on this data layout?

My assumption was the following:
```
200 USDC->DAI
[
 { path: USDCDAI30bps, inputAmount: 10, gasUsed: 150000 },
 { path: USDCDAI30bps, inputAmount: 50, gasUsed: 150000 },
 { path: USDCDAI5bps, inputAmount: 100, gasUsed: 150000 },
 { path: USDCDAI5bps, inputAmount: 200, gasUsed: 300000 },
]
```

From my reading, if we sell all 200 to USDCDAI5bps, the gas cost should be 300,000. From the accumulator is it calculating 450,000?

Reply:
I had misunderstood how how this worked, didn't realize that feel level was encoded into the path, so I thougth different unique paths meant different intermediate hops 😳 Updated to not double count the costs

Reply:
I found one issue with this approach last week. If the sampler aborts early/runs out of gas then `gasUsed` would be `0`, which would lead to an invalid gas estimation. Maybe it's better to revert to the original method, and just bump the gas depending on how many ticks were crossed? 🤔